src/components/Flyout/FlyoutWindow.part.tsx

```diff
@@ -366,7 +365,10 @@ export class FlyoutWindowInt extends React.Component<FlyoutWindowProps, FlyoutWi
   };
 
   componentDidMount() {
-    this.updateMeasurements();
+    if (this.flyoutContainer) {
+      this.updateMeasurements();
+      this.flyoutContainer.addEventListener('scroll', this.onScroll);
```

Comment:
Note - this is not a fix for the problem I described. With this solution you face two problems:#

1. If the DOM node changes during the lifetime of the component you no longer receive any updates (and disconnect the event from the wrong DOM node since your this.flyoutContainer is different in the end from the first version)
2. If no DOM node was available at mount (edge case and impl. dependent) then we never connected to the event

Reply:
Yes, but the DOM node can change during runtime.

As a simple (straight forward) example consider (it is one example, there are others where the node change is more hidden):

```jsx
class Hello extends React.Component {
  componentDidMount() {
	  console.log('cdm');
  }
  render() {
    return React.createElement(this.props.el, {
	    ref: el => console.log(el)
    }, `Hello ${this.props.name}`);
  }
}

class Carrier extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      el: 'i',
    };
  }
  componentDidMount() {
    setTimeout(() => {
    	this.setState({ el: 'b' });
    }, 2000);
  }
  render() {
    return <ul><li><Hello name="World" el={this.state.el} /></li></ul>;
  }
}

ReactDOM.render(
  <Carrier />,
  document.getElementById('container')
);
```

What you see here is 1. ref called (initial node) 2. did mount called 3. ref called (null) 4. ref called (new node).

TL;DR: You can go with the implicit mode as currently outlined, but for various robustness reasons I would propose to work against the ref node (if global node (e.g., this.flyoutContainer) is existing disconnect the listener(s), if local node (e.g., el) is existing connect the listener(s)) as it is resilient against all edge cases and not just one happy flow.